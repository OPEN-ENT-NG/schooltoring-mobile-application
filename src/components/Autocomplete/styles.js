import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  input: {
    backgroundColor: COLORS.WHITE
  },
  list: {
    backgroundColor: COLORS.WHITE,
    zIndex: 10,
    position: "relative"
  },
  item: {
    padding: 20
  }
});
