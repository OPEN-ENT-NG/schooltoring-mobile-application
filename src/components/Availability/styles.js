import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    paddingHorizontal: 30,
    paddingVertical: 15,
    flex: 1,
    justifyContent: "space-evenly"
  },
  item: {
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
    height: 60
  },
  itemSwitch: {
    marginRight: 20
  },
  itemText: {
    fontSize: 18
  },
  list: {
    flex: 1,
    justifyContent: "space-around"
  },
  title: {
    fontSize: 18,
    marginBottom: 40,
    flex: 0
  },
  buttonView: {
    flexDirection: "row",
    justifyContent: "flex-end",
    flex: 0,
    padding: 15
  }
});
