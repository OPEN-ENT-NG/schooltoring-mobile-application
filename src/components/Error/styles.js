import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column"
  },
  imageCritical: {
    width: 280,
    height: 301,
    position: "absolute",
    bottom: 0,
    left: -30
  },
  imageUncritical: {
    width: 262,
    height: 301,
    position: "absolute",
    bottom: 0,
    right: -30
  }
});

export default styles;
