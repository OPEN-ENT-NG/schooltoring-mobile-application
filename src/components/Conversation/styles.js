import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

const textColor = {
  color: COLORS.TEXT
};

export default StyleSheet.create({
  conversation: {
    flexDirection: "row",
    paddingHorizontal: 10
  },
  avatar: {
    alignSelf: "flex-start",
    flex: 1,
    alignItems: "center",
    paddingVertical: 20,
    marginHorizontal: 5
  },
  content: {
    flex: 4,
    justifyContent: "space-evenly",
    borderBottomColor: "#000",
    borderBottomWidth: StyleSheet.hairlineWidth
  },
  name: {
    ...textColor,
    fontWeight: "bold",
    fontSize: 20
  },
  message: {
    flexDirection: "row",
    justifyContent: "space-evenly"
  },
  icon: {
    fontSize: 20,
    marginRight: 10
  },
  text: {
    ...textColor,
    flex: 5,
    fontSize: 16,
    flexWrap: "wrap"
  },
  date: {
    flexGrow: 2,
    alignItems: "flex-end"
  },
  dateText: {
    ...textColor
  }
});
