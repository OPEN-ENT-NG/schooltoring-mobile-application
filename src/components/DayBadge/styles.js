import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  item: {
    justifyContent: "center",
    alignItems: "center",
    width: 30,
    height: 30,
    backgroundColor: COLORS.GREY
  },
  itemText: {
    flex: 1,
    color: COLORS.WHITE,
    fontSize: 20
  },
  available: {
    backgroundColor: COLORS.NEGATIVE
  }
});
