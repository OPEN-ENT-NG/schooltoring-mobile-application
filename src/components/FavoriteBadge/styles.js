import { StyleSheet, Platform } from "react-native";

import { COLORS } from "../../styles/common";

const shadow =
  Platform.OS === "android"
    ? {
        elevation: 4
      }
    : {
        shadowColor: COLORS.GREY,
        shadowOffset: { width: 2, height: 2 },
        shadowOpacity: 0.6,
        shadowRadius: 1
      };

export default StyleSheet.create({
  badge: {
    alignItems: "stretch",
    width: "50%",
    padding: 10
  },
  top: {
    aspectRatio: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: COLORS.SECONDARY_BACKGROUND,
    ...shadow
  },
  bottom: {
    backgroundColor: COLORS.BACKGROUND,
    ...shadow,
    maxHeight: 120
  },
  name: {
    textAlign: "center",
    color: COLORS.TEXT,
    fontWeight: "bold",
    fontSize: 16
  },
  buttons: {
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: "center",
    padding: 5
  },
  chatButton: {
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: COLORS.PRIMARY
  },
  loader: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: "rgba(255, 255, 255, 0.8)",
    elevation: 1000,
    zIndex: 5
  }
});
