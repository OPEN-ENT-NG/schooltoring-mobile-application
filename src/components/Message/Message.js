import React from "react";
import { View, Text } from "react-native";
import PropTypes from "prop-types";

import styles from "./styles";

import Avatar from "../Avatar/Avatar";

const Message = props => {
  const bubbleStyle = ({ color, direction }) => {
    let corner =
      direction === "left"
        ? {
            borderBottomRightRadius: 5
          }
        : { borderBottomLeftRadius: 5 };
    return { backgroundColor: color, ...corner };
  };

  return (
    <View
      style={[
        styles.message,
        { flexDirection: props.direction === "right" ? "row-reverse" : "row" }
      ]}
    >
      <View style={styles.avatar}>
        <Avatar
          size={60}
          src={`${global.config.auth.endpoint}${props.avatar}`}
        />
      </View>
      <View style={[styles.bubble, bubbleStyle(props)]}>
        <Text style={styles.text}>{props.message}</Text>
      </View>
    </View>
  );
};

Message.propTypes = {
  direction: PropTypes.oneOf(["right", "left"]),
  color: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
};

export default Message;
