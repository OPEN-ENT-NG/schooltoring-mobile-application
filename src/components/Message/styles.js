import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  message: {
    paddingVertical: 5,
    alignItems: "flex-end"
  },
  avatar: {
    paddingHorizontal: 10,
    justifyContent: "center"
  },
  bubble: {
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5,
    padding: 10,
    maxWidth: "50%",
    minWidth: "25%"
  },
  text: {
    color: COLORS.TEXT
  }
});
