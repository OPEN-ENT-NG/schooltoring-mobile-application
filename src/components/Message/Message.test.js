import React from "react";
import { shallow } from "enzyme";

import { COLORS } from "../../styles/common";
import state from "../../jest/state.json";
import Message from "./Message";

const message = {
  avatar: state.userinfo.avatar,
  color: COLORS.PRIMARY,
  message: "Test message"
};

const renderedLeftMessage = shallow(
  <Message
    direction="left"
    color={message.color}
    avatar={message.avatar}
    message={message.message}
  />
);

const renderedRightMessage = shallow(
  <Message
    direction="right"
    color={message.color}
    avatar={message.avatar}
    message={message.message}
  />
);

describe("Render", () => {
  test("Left message should render without crashing", () =>
    expect(renderedLeftMessage).toMatchSnapshot());

  test("Right message should render without crashing", () =>
    expect(renderedRightMessage).toMatchSnapshot());

  test(`Avatar source should be '${global.config.auth.endpoint}${
    message.avatar
  }'`, () => {
    expect(
      renderedLeftMessage
        .find("Avatar")
        .dive()
        .find("Image")
        .props().source.uri
    ).toEqual(`${global.config.auth.endpoint}${message.avatar}`);
  });

  test(`Message text should be '${message.message}'`, () => {
    expect(
      renderedLeftMessage.find("Text").contains(message.message)
    ).toBeTruthy();
  });
});
