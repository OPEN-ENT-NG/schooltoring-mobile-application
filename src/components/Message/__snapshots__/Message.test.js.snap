// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Left message should render without crashing 1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "paddingVertical": 5,
      },
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Component
    style={
      Object {
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  >
    <Avatar
      size={60}
      src="http://ent3-core.gdapublic.fr/workspace/document/49a54de3-51f8-464b-81bb-a377f508f382"
    />
  </Component>
  <Component
    style={
      Array [
        Object {
          "borderTopLeftRadius": 5,
          "borderTopRightRadius": 5,
          "maxWidth": "50%",
          "minWidth": "25%",
          "padding": 10,
        },
        Object {
          "backgroundColor": "#17b6b0",
          "borderBottomRightRadius": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#232323",
        }
      }
    >
      Test message
    </Text>
  </Component>
</Component>
`;

exports[`Render Right message should render without crashing 1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "paddingVertical": 5,
      },
      Object {
        "flexDirection": "row-reverse",
      },
    ]
  }
>
  <Component
    style={
      Object {
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  >
    <Avatar
      size={60}
      src="http://ent3-core.gdapublic.fr/workspace/document/49a54de3-51f8-464b-81bb-a377f508f382"
    />
  </Component>
  <Component
    style={
      Array [
        Object {
          "borderTopLeftRadius": 5,
          "borderTopRightRadius": 5,
          "maxWidth": "50%",
          "minWidth": "25%",
          "padding": 10,
        },
        Object {
          "backgroundColor": "#17b6b0",
          "borderBottomLeftRadius": 5,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#232323",
        }
      }
    >
      Test message
    </Text>
  </Component>
</Component>
`;
